core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -425,8 +425,7 @@ private[spark] object Utils extends Logging {
       conf: SparkConf,
       securityMgr: SecurityManager,
       hadoopConf: Configuration) {
-    val tempDir = getLocalDir(conf)
-    val tempFile =  File.createTempFile("fetchFileTemp", null, new File(tempDir))
+    val tempFile = File.createTempFile("fetchFileTemp", null, new File(targetDir.getAbsolutePath))
```

Comment:
Yeah, we probably can just use `targetDir` directly; I guess I overlooked this when reviewing the patch.